onyo/commands/tests/test_new.py

```diff
@@ -86,6 +86,98 @@ def test_new_top_level(repo: OnyoRepo) -> None:
     fsck(repo)
 
 
+@pytest.mark.repo_dirs(*directories)
+def test_new_from_root_with_absolute_paths(repo: OnyoRepo) -> None:
+    """
+    Test that `onyo new --path <path>` can create an asset with absolute paths
+    while in the root of the repository.
+    """
+    file = 'laptop_apple_macbookpro.faux'
+
+    for d in directories:
+        # build an absolute path for a new asset
+        new_file = Path(f"{d}/{file}").resolve()
+
+        ret = subprocess.run(['onyo', 'new', '--yes', '--path', new_file],
+                             capture_output=True, text=True)
+
+        # verify correct output
+        assert "The following will be created:" in ret.stdout
+        assert str(d) in ret.stdout
+        assert str(new_file) in ret.stdout
+        assert not ret.stderr
+        assert ret.returncode == 0
+
+    # verify that all new assets exists and the repository is in a clean state
+    assert len(repo.asset_paths) == len(directories)
+    fsck(repo)
+
+
+@pytest.mark.repo_dirs(*directories)
+def test_new_sub_dir_absolute_path(repo: OnyoRepo) -> None:
+    """
+    Test that `onyo new --path <path>` can create an asset in sub-directories of
+    the repository while the cwd is inside a different sub-directory.
+    """
+    file = 'laptop_apple_macbookpro.faux'
+
+    for cwd in directories:
```

Comment:
This is expensive. Why test every single combination of directories?

Take one of the `directories` for CWD and another one for the asset. That's what you want to test (according to the docstring). What does repeating the test with different directories add, that wasn't already covered?

Reply:
Hm. While this is a little better, that's not what I meant. You have one dir for CWD and and asset in another dir. Fine. That's what you want to test. What does it add to also run `new` for other assets in even more dirs? What does that cover that is not covered by chdir into `directories[0]` and run new on an asset in `directories[1]`? How is the rest of the directories different? Dealing with all sorts of paths (several levels, unicode, etc.) has nothing to do with the stated purpose of this test, but with general path handling. That's tested elsewhere.